IPython/core/interactiveshell.py

```diff
@@ -3309,7 +3312,16 @@ def error_before_exec(value):
             assert self.history_manager is not None
             # Write output to the database. Does nothing unless
             # history output logging is enabled.
-            self.history_manager.store_output(self.execution_count)
+            hm = self.history_manager
+            hm.store_output(self.execution_count)
+            exec_count = self.execution_count
+
+            # Capture MIME outputs and exceptions
+            if result.result:
+                hm.output_mime_bundles[exec_count] = deepcopy(result.result)
```

Comment:
I wonder if we should prefer to:
a) perform the conversion to mime bundle here (call `self.shell.display_formatter.format`), or
b) store a deep copy (as now).

I think (a) would be better because otherwise user with a large data frame (say few GB) will quickly exhaust their RAM, unless.